File: 백준/Gold/2436. 공약수/공약수.py
def euclidean(a: int, b: int):
    # 큰수 a
    # 작은 수 b
    while a % b != 0:
        temp = b
        b = a % b
        a = temp
    
    return b

gcd, lcm = map(int, input().split())
ab = gcd * lcm
A = gcd
B = lcm

for a in range(gcd, int(ab ** 0.5) + 1, gcd):
    if ab % a == 0:
        b = ab // a
        if euclidean(a, b) == gcd:
            if A + B > a + b:
                A = a
                B = b

print(A, B)

File: 백준/Gold/2436. 공약수/공약수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] s = br.readLine().split(" ");
        solution(Integer.parseInt(s[0]), Integer.parseInt(s[1]));
    }

    static void solution(int gcd, int lcm) {
        long ab = (long) gcd * lcm;
        int A = gcd;
        int B = lcm;
        for (int a = gcd; a <= Math.sqrt(ab); a += gcd) {
            if (ab % a == 0) {
                int b = (int) (ab / a);
                if (getGCD(a, b) == gcd) {
                    if (A + B > a + b) {
                        A = a;
                        B = b;
                    }
                }
            }
        }
        System.out.println(A + " " + B);
    }

    static long getGCD(int a, int b) {
        return b == 0 ? a : getGCD(b, a % b);
    }
}